### ISSUE TITLE
No longer sending real errors to `_error.js` on production builds.

### ISSUE THREAD

**Issue by gcpantazis** [zeit/next.js#1814]
(Copying from #1800 since this is a new issue as a result of that cc/ @arunoda)

#1800 introduced a "fake error" object that's sent to the _error view on production. This creates a feature divergence on dev vs. production, and removed the possibility for tracking errors in `getInitialProps`.

My use case is that I want to send that error message to our logs, and the _error.js page is the best single place I see to capture errors that break next on both the client and server. i.e.:

```
export default class ErrorPage extends React.Component {
    static getInitialProps({res, jsonPageRes, err}) {
        if (err) {
            track.metric('render_error');
            // Utility method that serializes JS errors on client and server, sends them to logs
            track.error(err);
        }

        return {}
    }

    render() {
        return (
            <div>...</div>
        )
    }
}
```
If we're not consistently sending the error object to the error page, do we have a better way to grab and track them?

**Comment by arunoda** [comment]
Answered here: https://github.com/zeit/next.js/pull/1800#issuecomment-297608767

**Comment by gcpantazis** [comment]
Okay, a few questions / points:

1) These aren't necessarily *just* server errors, since the apps are universally rendered.
2) Since this was removed, is there any one place in the application to predictably get all errors from `next`? *Ideally* it'd be the same server and client. I've gone through the source and it isn't really clear how errors are supposed to be handled.

**Comment by sethsamuel** [comment]
This is causing some serious problems for us, since Googlebot is triggering a mysterious client-side error when trying to render our pages, and we have not way of debugging it.

**Comment by gcpantazis** [comment]
Just wanted to bump this up; I'm still relying on a fork to get this functionality in. I'm guessing there isn't a better prescribed way to track errors in production yet since the `serializeErrors` method in server/render is still gated on `dev`.

I'm happy to send in a PR if we can agree on the right way to proceed; a next config option to send along the errors in production, perhaps?

cc/ @sergiodxa @arunoda @timneutkens

**Comment by gcpantazis** [comment]
This was fixed as of `4.1.4`, closing

**Comment by timneutkens** [comment]
Awesome @gcpantazis
